tests/backend/integration/api/projects/test_teams.py

```diff
@@ -0,0 +1,238 @@
+        """
+        response = self.client.post(
+            self.single_project_team_url,
+            json={"role": TeamRoles.MAPPER.name},
+            headers={"Authorization": self.test_user_session_token},
+        )
+        response_body = response.get_json()
+        self.assertEqual(response.status_code, 403)
+        self.assertEqual(
+            response_body["Error"], "User is not an admin or a manager for the team"
+        )
+        self.assertEqual(response_body["SubCode"], "UserPermissionError")
+
+    def test_assign_team_to_non_existent_project_fails(self):
+        """
+        Test that endpoint returns 404 when admin assigns a team to a non-existent project
+        """
+        response = self.client.post(
+            f"/api/v2/projects/99/teams/{self.test_team.id}/",
+            json={"role": TeamRoles.MAPPER.name},
+            headers={"Authorization": self.test_author_session_token},
+        )
+        response_body = response.get_json()
+        self.assertEqual(response.status_code, 404)
+        self.assertEqual(response_body["Error"], "No Project Found")
+        self.assertEqual(response_body["SubCode"], "NotFound")
+
+    def test_assign_team_to_project_by_admin_passes(self):
+        """
+        Test that endpoint returns 201 when admin successfully assigns a team to a project
+        """
+        response = self.client.post(
+            self.single_project_team_url,
+            json={"role": TeamRoles.MAPPER.name},
+            headers={"Authorization": self.test_author_session_token},
+        )
+        response_body = response.get_json()
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual(
+            response_body["Success"],
+            f"Team {self.test_team.id} assigned to project {self.test_project.id} with role MAPPER",
+        )
+
+    # patch
+    def test_update_team_role_by_unauthenticated_user_fails(self):
+        """
+        Test that endpoint returns 401 when unauthenticated user updates project team role
+        """
+        response = self.client.patch(
+            self.single_project_team_url, json={"role": TeamRoles.MAPPER.name}
+        )
+        response_body = response.get_json()
+        self.assertEqual(response.status_code, 401)
+        self.assertEqual(response_body["SubCode"], "InvalidToken")
+
+    def test_update_team_role_by_non_admin_fails(self):
+        """
+        Test that endpoint returns 403 when non admin updates project team role
+        """
+        response = self.client.patch(
+            self.single_project_team_url,
+            json={"role": TeamRoles.MAPPER.name},
+            headers={"Authorization": self.test_user_session_token},
+        )
+        response_body = response.get_json()
+        self.assertEqual(response.status_code, 403)
+        self.assertEqual(response_body["Error"], "User is not a manager of the project")
+        self.assertEqual(response_body["SubCode"], "UserPermissionError")
+
```

Comment:
Same here a test to determine whether a 400 is raised when an invalid team role is passed appears to be missing.